scriptmodules/ports/solarus.sh

```diff
@@ -20,53 +20,67 @@ function depends_solarus() {
+    tar xzf zelda-xd2-1.0.6.tar.gz
 }
 
 function build_solarus() {
     mkdir build
     cd build
-    cmake .. -DCMAKE_INSTALL_PREFIX="$md_inst"
+    cmake .. -DCMAKE_INSTALL_PREFIX="$md_inst" -DSOLARUS_GUI=OFF
     make
-    cd ../zsdx-1.10.3
+    cd ../zsdx-1.11.0
     cmake . -DCMAKE_INSTALL_PREFIX="$md_inst"
     make
-    cd ../zsxd-1.10.3
+    cd ../zsxd-1.11.0
     cmake . -DCMAKE_INSTALL_PREFIX="$md_inst"
     make
-    cd ../zelda-roth-se-1.0.8
+    cd ../zelda-roth-se-1.1.0
+    cmake . -DCMAKE_INSTALL_PREFIX="$md_inst"
+    make
+    cd ../zelda-xd2-1.0.6
     cmake . -DCMAKE_INSTALL_PREFIX="$md_inst"
     make
     md_ret_require=(
-        "$md_build/build/solarus_run"
-        "$md_build/zsdx-1.10.3/data.solarus"
-        "$md_build/zsxd-1.10.3/data.solarus"
-        "$md_build/zelda-roth-se-1.0.8/data.solarus"
+        "$md_build/build/solarus-run"
+        "$md_build/zsdx-1.11.0/data.solarus"
+        "$md_build/zsxd-1.11.0/data.solarus"
+        "$md_build/zelda-roth-se-1.1.0/data.solarus"
+        "$md_build/zelda-xd2-1.0.6/data.solarus"
     )
 }
 
 function install_solarus() {
     cd build
     make install
-    cd ../zsdx-1.10.3/
+    cd ../zsdx-1.11.0/
     make install
-    cd ../zsxd-1.10.3/
+    cd ../zsxd-1.11.0/
     make install
-    cd ../zelda-roth-se-1.0.8/
+    cd ../zelda-roth-se-1.1.0/
+    make install
+    cd ../zelda-xd2-1.0.6/
     make install
 }
 
 function configure_solarus() {
-    addPort "$md_id" "zsdx" "Solarus Engine - Zelda Mystery of Solarus DX" "LD_LIBRARY_PATH=$md_inst/lib $md_inst/bin/solarus_run $md_inst/share/solarus/zsdx/"
-    addPort "$md_id" "zsxd" "Solarus Engine - Zelda Mystery of Solarus XD" "LD_LIBRARY_PATH=$md_inst/lib $md_inst/bin/solarus_run $md_inst/share/solarus/zsxd/"
-    addPort "$md_id" "zelda_roth_se" "Solarus Engine - Zelda Return of the Hylian SE" "LD_LIBRARY_PATH=$md_inst/lib $md_inst/bin/solarus_run $md_inst/share/solarus/zelda_roth_se/"
+		addPort "$md_id" "zsdx" "Solarus Engine - Zelda Mystery of Solarus DX" "LD_LIBRARY_PATH=/usr/lib $md_inst/bin/solarus-run $md_inst/share/solarus/zsdx/"
+    	addPort "$md_id" "zsxd" "Solarus Engine - Zelda Mystery of Solarus XD" "LD_LIBRARY_PATH=/usr/lib $md_inst/bin/solarus-run $md_inst/share/solarus/zsxd/"
+    	addPort "$md_id" "zelda_roth_se" "Solarus Engine - Zelda Return of the Hylian SE" "LD_LIBRARY_PATH=/usr/lib $md_inst/bin/solarus-run $md_inst/share/solarus/zelda_roth_se/"
+    	addPort "$md_id" "zelda_xd2_mercuris_chess" "Solarus Engine - Zelda XD2 Mercuris Chess" "LD_LIBRARY_PATH=/usr/lib $md_inst/bin/solarus-run $md_inst/share/solarus/zelda_xd2_mercuris_chess/"
 
+    if isPlatform "tinker"; then
+    	sudo mv /opt/retropie/port/solarus/lib/arm-linux-gnueabihf/* /usr/lib/arm-linux-gnueabihf/
+    	sudo rm -rf /opt/retropie/port/solarus/lib/arm-linux-gnueabihf
+		ln -sf /usr/lib/arm-linux-gnueabihf/libsolarus.so "$md_inst/lib/libsolarus.so"
```

Comment:
wrong indent

Reply:
I don't like this library code, should be doable using LD_LIBRARY_PATH without having to copy or symlink libs.